src/strapp/dramatiq/base.py

```diff
@@ -0,0 +1,163 @@
+from dataclasses import dataclass, field
+from typing import Callable, Optional, Union
+
+import dramatiq
+from dramatiq.brokers.redis import RedisBroker
+from dramatiq.results import Results
+from dramatiq.results.backends import RedisBackend
+
+try:
+    from strapp.dramatiq.datadog import DatadogMiddleware
+except:  # noqa: E722
+    pass
+
+
+def configure(
+    *, redis_dsn=None, enable_datadog_middleware: bool = False, env: str = None
+) -> RedisBroker:
+    """Configure a Redis broker.
+
+    Both the worker itself, as well as any code which wants to `enqueue` work, should call this
+    function at startup.
+    """
+
+    backend = RedisBackend(url=redis_dsn)
+    broker = RedisBroker(url=redis_dsn)
+    broker.add_middleware(Results(backend=backend))
+
+    if enable_datadog_middleware:
+        broker.add_middleware(DatadogMiddleware(env=env))
+
+    dramatiq.set_broker(broker)
+
+    return broker
+
+
+class PreparedActor:
+    """Prepared configuration for a Dramatiq Actor.
+
+    The default actor implementation does not require an explicit name and uses the function name
+    by default. In order to safely use `enqueue` for tasks, being explicit about the name seems safer.
+
+    Keep in mind that actor names must be globally unique.
+
+    dramatiq default timeout is 10 minutes, but we default to 1 hour for greater flexibility.
+
+    The default max_retries for dramatiq is 5, but we default to 0. If you notice your task
+    failing due to transient network behavior then consider upping it and modifying backoff parameters.
+    https://dramatiq.io/guide.html#message-retries
+    """
+
+    def __init__(
+        self,
+        fn: Callable,
+        *,
+        actor_name: Optional[str] = None,
+        broker: dramatiq.Broker = None,
+        queue_name: str = "default",
+        store_results: bool = False,
+        time_limit_ms: int = 1000 * 60 * 60,
+        max_retries: int = 0,
+        **options,
+    ):
+        self.fn = fn
+        self.actor_name = actor_name
+        self.broker = broker
+        self.queue_name = queue_name
+        self.store_results = store_results
+        self.time_limit_ms = time_limit_ms
+        self.max_retries = max_retries
+        self.options = options
+
+    def register(self, broker: dramatiq.Broker = None) -> dramatiq.Actor:
+        """Register an actor to a broker."""
+
+        actor_decorator = dramatiq.actor(
+            # arguments for dramatiq.Actor
+            actor_name=self.actor_name or self.fn.__name__,
+            broker=broker or self.broker,
+            queue_name=self.queue_name,
+            # options passed to middleware
+            store_results=self.store_results,
+            time_limit=self.time_limit_ms,
+            max_retries=self.max_retries,
+            **self.options,
+        )
+
+        return actor_decorator(self.fn)
+
+
+def enqueue(task_name, *, dramatiq_queue="default", broker=None, **kwargs) -> dramatiq.Message:
```

Comment:
Up above you use `queue_name` and here you use `dramatiq_queue` -- i would just suggest to keep it consistent with maybe slight preference to `queue_name` since thats what dramatiq uses